lib/github_changelog_generator/octo_fetcher.rb

```diff
@@ -373,14 +376,17 @@ def print_empty_line
     def fetch_github_token
       env_var = @options[:token] ? @options[:token] : (ENV.fetch CHANGELOG_GITHUB_TOKEN, nil)
 
-      Helper.log.warn NO_TOKEN_PROVIDED unless env_var
-
-      token = env_var.split(',')[@token_idx]
-      if token
-        Helper.log.warn "[OLD] Using token[#{@token_idx}]=#{token[0..5]}***********************************"
-        @token_idx = @token_idx + 1
-      else
-        sys_abort("No more github tokens to try, exiting")
+      token = env_var
+      if env_var.nil?
+        Helper.log.warn NO_TOKEN_PROVIDED
+      elsif env_var.include? ","
+        token = env_var.split(',')[@token_idx]
+        if token
+          Helper.log.warn "[OLD] Using token[#{@token_idx}]=#{token[0..5]}***********************************"
+          @token_idx = @token_idx + 1
+        else
+          sys_abort("No more github tokens to try, exiting")
+        end
```

Comment:
FYI the above section was added by me to experiment with using multi-token in the same run, I'd like to leave it here for now. We can clean it up later if this PR improves the changelog generator stability :-)

Reply:
@unicolet Nice, I like that there's native multi-tokens in it now, can drop the `changelog || changelog` block in the Jenkins job.

Reply:
@dawn-minion question is, will it work? :sweat_smile: Last I tried it was not quite the success I hoped, but maybe I should give it another try now